server/stt/stt_planning_ml.py

```diff
@@ -124,7 +139,227 @@ async def get_coverage_details(
             # Return ``None`` so this coverage isn't added to the Planning item
             return None
 
-        return await super().get_coverage_details(news_coverage_elt, item, original)
+        coverage = await super().get_coverage_details(news_coverage_elt, item, original)
+        if coverage is not None:
+            # Parse STT-specific fields for all coverages
+            self.parse_stt_coverage_fields(news_coverage_elt, coverage)
+        return coverage
+
+    def parse_stt_coverage_fields(
+        self, news_coverage_elt: Element, coverage: Dict[str, Any]
+    ):
+        """Parse STT-specific coverage fields from XML"""
+        planning_elt = news_coverage_elt.find(self.qname("planning"))
+        if planning_elt is None:
+            return
+
+        # Initialize coverage structure with proper fields
+        coverage.setdefault("planning", {})
+        coverage["planning"].setdefault("fields", {})
+        coverage["planning"].setdefault("subject", [])
+
+        # Parse all fields efficiently in single iterations
+        self.parse_all_subject_fields(planning_elt, coverage)
+        self.parse_non_subject_fields(planning_elt, coverage)
+
+    def parse_all_subject_fields(self, planning_elt: Element, coverage: Dict[str, Any]):
+        """Parse all subject fields in a single iteration"""
+        for subject_elt in planning_elt.findall(self.qname("subject")):
+            qcode = subject_elt.get("qcode", "")
+            value_elt = subject_elt.find(self.qname("value"))
+            value_text = (
+                value_elt.text.strip()
+                if value_elt is not None and value_elt.text
+                else ""
+            )
+
+            if qcode.startswith("sttworkstatus:"):
+                self.parse_coverage_status(subject_elt, coverage)
+            elif qcode == "sttinternaltext":
+                coverage["internal_note"] = value_text
+            elif qcode.startswith("sttimagetypename:"):
+                self.parse_picture_type(subject_elt, coverage)
+            elif qcode.startswith("sttphotoaware:"):
+                self.parse_photographer_awareness(subject_elt, coverage)
+            elif qcode == "sttentryinfo":
+                coverage["planning"]["fields"]["sttregistrationinfo"] = value_text
```

Comment:
Please check the schema for [fields](https://github.com/superdesk/superdesk-planning/blob/develop/server/planning/planning/planning_schema.py#L134), as it is not a dictionary